src/modules/esl-anchornav/core/esl-anchornav.ts

```diff
@@ -0,0 +1,224 @@
+export class ESLAnchornav extends ESLBaseElement {
+  public static override is = 'esl-anchornav';
+  public static _renderers: Map<string, ESLAnchornavRender> = new Map();
+
+  /** Gets renderer by name */
+  public static getRenderer(name: string): ESLAnchornavRender | undefined {
+    return this._renderers.get(name);
+  }
+
+  /** Sets renderer */
+  public static setRenderer(renderer: ESLAnchornavRender): void;
+  public static setRenderer(name: string, renderer: ESLAnchornavRender): void;
+  public static setRenderer(name: string | ESLAnchornavRender, renderer?: ESLAnchornavRender): void {
+    if (typeof name !== 'string') return this.setRenderer('default', name);
+    if (typeof name === 'string' && renderer) this._renderers.set(name, renderer);
+  }
+
+  @prop('esl:anchornav:activechanged') public ACTIVECHANGED_EVENT: string;
+  @prop('esl:anchornav:updated') public UPDATED_EVENT: string;
+  @prop('[esl-anchor]') protected ANCHOR_SELECTOR: string;
+
+  /** Item renderer which is used to build inner markup */
+  @attr({defaultValue: 'default', name: 'renderer'}) public rendererName: string;
+
+  protected _active: ESLAnchorData;
+  protected _anchors: ESLAnchorData[] = [];
+  protected _items: Map<string, Element> = new Map();
+  protected _offset: number;
+
+  /** Active anchor */
+  public get active(): ESLAnchorData {
+    return this._active;
+  }
+  public set active(value: ESLAnchorData) {
+    if (this._active === value) return;
+    this._active = value;
+    this._onActiveChange(value);
+  }
+
+  /** Anchors list */
+  protected get $anchors(): HTMLElement[] {
+    return this._anchors.map(({$anchor}) => $anchor);
+  }
+
+  /** Anchornav offset */
+  public get offset(): number {
+    return this._offset || 0;
+  }
+  public set offset(value: number) {
+    if (this._offset === value) return;
+    this._offset = value;
+    memoize.clear(this, '$viewport');
+    this.$$on(this._onAnchorIntersection);
+  }
+
+  /** Anchornav item renderer */
+  protected get itemRenderer(): ESLAnchornavRender | undefined {
+    return ESLAnchornav.getRenderer(this.rendererName) || ESLAnchornav.getRenderer('default');
+  }
+
+  /** Anchornav items container */
+  @memoize()
+  protected get $itemsArea(): HTMLElement {
+    const $provided = this.querySelector<HTMLElement>(`[${this.baseTagName}-items]`);
+    if ($provided) return $provided;
+    const $container = document.createElement('div');
+    $container.setAttribute(this.baseTagName + '-items', '');
+    this.appendChild($container);
+    return $container;
+  }
+
+  /** Anchornav viewport (root element for IntersectionObservers checking visibility) */
+  @memoize()
+  protected get $viewport(): Element | undefined {
+    return getViewportForEl(this);
+  }
+
+  @ready
+  protected override connectedCallback(): void {
+    super.connectedCallback();
+    this.init();
+  }
+
+  /** Initializes the component */
+  public init(): void {
+    this._onAnchornavRequest();
+  }
```

Comment:
Do we need extra init method? it seems to be it's a little bit excessive

Reply:
Just for case when need to recollect anchors. But it is an infrequent case and the event could be used for this.
removed